transform.go

```diff
@@ -103,11 +103,19 @@ func transformImage(m image.Image, opt Options) image.Image {
 
 	// never resize larger than the original image
 	if !opt.ScaleUp {
-		if w > imgW {
-			w = imgW
+		if w >= imgW {
+			if h == 0 {
```

Comment:
Couldn't that be simplified to:

```go
if (w == imgW || w == 0) && (h == imgH || h == 0) {
    w, h = 0, 0
}
```

Basically, if both height and width or either zero or the original value, then no resize is needed.

Yeah, pulling this out into a separate func would certainly make testing it a bit easier.  Though I don't think that func would need to be exported (so `resizeParams`), and using the name `ok` seems confusing here... maybe just `resize` or something?